### PR TITLE
feat(CRUD accommodation facilities)Travel admins should be able to CR…

### PR DESCRIPTION
**Pull request by munyanezaarmel**
###  What does this PR do?

 Travel admins should be able to CRUD accommodation facilities

### Description of Task to be completed?

have the following endpoints working

-  **GET /api/v1/ accomodation/**
-  **POST /api/v1/ accomodation/**
- **GET /api/v1/ accomodation/**
- **GET /api/v1/ accomodation/<accomodation_id>**
- **PUT /api/v1/ accomodation/<accomodation_id>**
- **DELETE /api/v1/ accomodation/<accomodation_id>**
- **GET /api/ v1/room/**
- **POST /api/ v1/room/**
- **GET /api/ v1/room/**
- **GET /api/ v1/room/<room_id>**
- **PUT /api/v1/ room/<room_id>**
- **DELETE /api/v1/ room/<room_id>**
- **POST /api/v1/country**
- **GET/api/v1/country**
- **POST /api/v1/location/create**
- **GET /api/v1/location**
- **DELETE/api/v1/location/delete/<locationId>**
- **PATCH /api/v1/location/update/<locationId>**

### How should this be manually tested?
after cloning this repo cd into it and run npm run start

### - using swagger documentation 
**go to http://localhost:3000/api-docs**

### -  using postman test every endpoint above with this header:

 key: content-type value : application/json

###  Any background context you want to provide?
- travel should first create country->location->accommodation-room
### What are the relevant pivotal tracker stories?
[#182201574](https://www.pivotaltracker.com/story/show/182201574)

### CODE REVIEW

**Review by kalisaNkevin — CHANGES_REQUESTED** [review]
rebase needed

**Review by ishimweeli — DISMISSED** [review]

**Review by kalisaNkevin — DISMISSED** [review]
running successfuly

File: .env.example [hunk 1]
```diff
@@ -25,3 +31,4 @@ JWT_COOKIE_EXPIRES_IN=time
 
 NODE_ENV=test | development | production
 
+NODE_ENV=test
```
Comment (didasmbalanya) [inline .env.example]: remove

File: src/services/locationServices.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { Accomodation, Location, Country } from '../database/models';
+
+class locationServices {
+  static create = async (location) => {
+    return Location.create(location);
+  };
+
+  static getSingleLocation = async (id) => {
+    console.log('############# id')
```
Comment (didasmbalanya) [inline src/services/locationServices.js]: remove console.log

File: src/controllers/accomodation.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+import LocationService from '../services/locationServices';
+import * as ApplicationError from '../utils/errors/applicatioErrors';
+import { Accommodation } from '../database/models';
+
+class AccomodationController {
+  static createAccommodation = async (req, res, next) => {
+    try {
+      const location = await LocationService.getSingleLocation(req.body.locationId);
+      if (!location) {
+        return ApplicationError.internalServerError(
```
Comment (didasmbalanya) [inline src/controllers/accomodation.js]: this is error 404, not 500 - internal server error

File: src/controllers/accomodation.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        imageResponse.forEach((image) => {
+          return urls.push(image.url);
+        });
+        req.body = {
+          ...req.body,
+          accomodationImage: urls,
+        };
+      }
+      const createdAccommodation = await AccommodationService.createAccommodation(req.body);
+      console.log(req.body)
```
Comment (didasmbalanya) [inline src/controllers/accomodation.js]: remove console.log

File: test/country.spec.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    it('It should get all countries', async () => {
+        const res = await chai
+          .request(app)
+          .get('/api/v1/country')
+          .send(country);
+        expect(res).to.have.status(200);
+      });
+  
+  })
```
Comment (didasmbalanya) [inline test/country.spec.js]: This is tests for accomodaton ! not validation

File: src/routes/accomodationRouter.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import express from 'express';
+import accomodationController from '../controllers/accomodation';
+import upload from '../utils/helpers/fileUploads/multer';
+
+const accomodationRouter = express.Router();
+
+accomodationRouter.post(
+  '/v1/accomodation/create',
```
Comment (didasmbalanya) [inline src/routes/accomodationRouter.js:11]: add a middleware to validate all endpoints that create data. use JOI validation!

File: src/database/migrations/20220610110828-create-accomodation.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      },
+      accomodationName: {
+        type: Sequelize.STRING
+      },
+      accomodationDescription: {
+        type: Sequelize.STRING
+      },
+      locationId: {
+        type: Sequelize.INTEGER
+      },
```
Comment (didasmbalanya) [inline src/database/migrations/20220610110828-create-accomodation.js]: missing relationship to location table

File: src/database/migrations/20220610111626-create-room.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    await queryInterface.createTable('Rooms', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      accomodationId: {
+        type: Sequelize.INTEGER
+      },
```
Comment (didasmbalanya) [inline src/database/migrations/20220610111626-create-room.js]: add relationship

File: src/database/models/accomodation.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  }
+  Accomodation.init({
+    id: {
+      allowNull: false,
+      autoIncrement: true,
+      primaryKey: true,
+      type: DataTypes.INTEGER
+    },
+    accomodationName: DataTypes.STRING,
+    accomodationDescription: DataTypes.STRING,
```
Comment (didasmbalanya) [inline src/database/models/accomodation.js]: not null

Comment (didasmbalanya) [inline src/database/models/accomodation.js]: not null

File: src/index.js [hunk 10]
```diff
@@ -1,6 +1,5 @@
-
-import express from 'express'
-import babel from "@babel/core";
+import express from 'express';
+import babel from '@babel/core';
 import swaggerDocs from '../public/api-docs/swagger.js';
 import { sequelize } from "./database/models"
 import dotenv from 'dotenv'
```
Comment (didasmbalanya) [inline src/index.js]:
```suggestion
import 'dotenv/config'
```

**Review by didasmbalanya — CHANGES_REQUESTED** [review]
- Make sure your data is validated before being engaged in any controller
- Make sure your models and migrations match both in relations and types
- Testing should also include a lot of the data not just status code. more test cases needed

File: src/routes/accomodationRouter.js (hunk 6 above)
Comment (munyanezaarmel) [reply src/routes/accomodationRouter.js:11]: done

File: src/database/models/accomodation.js (hunk 9 above)
Comment (munyanezaarmel) [reply src/database/models/accomodation.js]: done

Comment (munyanezaarmel) [reply src/database/models/accomodation.js]: done

File: src/database/migrations/20220610111626-create-room.js (hunk 8 above)
Comment (munyanezaarmel) [reply src/database/migrations/20220610111626-create-room.js]: done

File: src/database/migrations/20220610110828-create-accomodation.js (hunk 7 above)
Comment (munyanezaarmel) [reply src/database/migrations/20220610110828-create-accomodation.js]: done

File: src/controllers/accomodation.js (hunk 4 above)
Comment (munyanezaarmel) [reply src/controllers/accomodation.js]: done

File: src/controllers/accomodation.js (hunk 3 above)
Comment (munyanezaarmel) [reply src/controllers/accomodation.js]: done

File: src/services/locationServices.js (hunk 2 above)
Comment (munyanezaarmel) [reply src/services/locationServices.js]: done

File: test/country.spec.js (hunk 5 above)
Comment (munyanezaarmel) [reply test/country.spec.js]: done

**Review by Julienish14 — COMMENTED** [review]
I think a country must be unique, you have unused findOne method in countryServise. please use it to avoid country duplication in database

Fixed check again

**Comment by kalisaNkevin** [comment]
work on the tests

**Review by kalisaNkevin — DISMISSED** [review]

**Review by Julienish14 — APPROVED** [review]

**Review by Julienish14 — DISMISSED** [review]
I have a little question? aren't this supposed to be protected?

**Review by ishimweeli — APPROVED** [review]

**Review by ishimweeli — DISMISSED** [review]

**Review by kalisaNkevin — DISMISSED** [review]

**Review by kalisaNkevin — DISMISSED** [review]

File: src/controllers/accomodation.js [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+        };
+      }
+      const createdAccommodation = await AccommodationService.createAccommodation(req.body);
+      return res.status(201).json({
+        status: '201',
+        message: 'Accommodation added successfully',
+        payload: createdAccommodation,
+      });
+    } catch (error) {
+      console.log(error);
```
Comment (Ntare22) [inline src/controllers/accomodation.js]: Please remove the console logs.

File: src/controllers/accomodation.js [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+        req.body
+      );
+
+      res.status(200).json({
+        status: 200,
+        message: 'accommodation updated successfully',
+        payload: updatedAccommodation,
+      });
+    } catch (error) {
+      console.log(error);
```
Comment (Ntare22) [inline src/controllers/accomodation.js]: Remove this console too.

**Review by Ntare22 — DISMISSED** [review]
Please remove the console.logs in your PR.

File: src/controllers/accomodation.js (hunk 12 above)
Comment (munyanezaarmel) [reply src/controllers/accomodation.js]: done

**Review by Francoise02 — DISMISSED** [review]

**Review by kalisaNkevin — DISMISSED** [review]

**Review by Francoise02 — APPROVED** [review]

**Review by uwishema11 — APPROVED** [review]

**Review by Julienish14 — APPROVED** [review]

**Review by kalisaNkevin — APPROVED** [review]

**Review by ishimweeli — APPROVED** [review]
working well

**Review by didasmbalanya — APPROVED** [review]

**Pull request merged**